audio_alsa.c

```diff
@@ -560,10 +560,22 @@ static int actual_open_alsa_device(int do_auto_setup) {
     return ret;
   }
 
-  ret = snd_pcm_hw_params_set_channels(alsa_handle, alsa_params, 2);
+  int channels = 2;
+  if (config.playback_mode == ST_one_channel) {
+    channels = 1;
+  } else {
+    ret = snd_pcm_hw_params_test_channels(alsa_handle, alsa_params, channels);
+    if (ret <0) {
+      debug(1, "Failed to get 2 channels revert back to 1");
+      channels = 1;
+      config.playback_mode = ST_one_channel;
+    }
+  }
+
+  ret = snd_pcm_hw_params_set_channels(alsa_handle, alsa_params, channels);
   if (ret < 0) {
-    die("audio_alsa: Channels count (2) not available for device \"%s\": %s", alsa_out_dev,
-        snd_strerror(ret));
+    die("audio_alsa: Channels count (%d not available for device \"%s\": %s",
```

Comment:
Missing a bracket here?

Reply:
Missing parentheses after the %d. got it.